src/compute-client/src/controller/instance.rs

```diff
@@ -539,12 +539,23 @@ impl<T: ComputeControllerTimestamp> Instance<T> {
     ///
     /// This also returns `true` in case this cluster does not have any
     /// replicas.
-    pub fn all_collections_hydrated(&self) -> bool {
+    pub fn all_collections_hydrated_filtered(
+        &self,
+        filter_replica_ids: Option<Vec<ReplicaId>>,
```

Comment:
I think it would be helpful to update the doc-comment to indicate what this list is. It's not immediately clear if we ignore replicas in the list or we only look at replicas in this list and how the `Option` affects things.